widget/table.go

```diff
@@ -0,0 +1,299 @@
+package widget
+
+import (
+	"image/color"
+	"math"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/theme"
+)
+
+// Table widget is a grid of items that can be scrolled and a cell selected.
+// It's performance is provided by caching cell templates created with NewCell and re-using them with UpdateCell.
+// The size of the content rows/columns is returned by the DataSize callback.
+type Table struct {
+	BaseWidget
+
+	DataSize       func() (int, int)
+	NewCell        func() fyne.CanvasObject
+	UpdateCell     func(row int, col int, template fyne.CanvasObject)
+	OnCellSelected func(row int, col int)
+
+	cells                       *tableCells
+	scroll                      *ScrollContainer
+	SelectedRow, SelectedColumn int
```

Comment:
Could the exported fields be moved above the unexported ones?

Reply:
Yes. Though worth noting that ordering of fields in struct is not something we have standardised.

Reply:
Yup, something we should add to the consistency wiki